### PR TITLE
[aot] Simplify a check to avoid calling into the assembly loading code while loading the aot image for corlib.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/16380.

### CODE REVIEW

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -299,9 +299,7 @@ load_image (MonoAotModule *amodule, int index, MonoError *error)
 	 * current AOT module matches the wanted name and guid and just return
 	 * the AOT module's assembly.
 	 */
-	if (mono_asmctx_get_kind (&amodule->assembly->context) == MONO_ASMCTX_INDIVIDUAL &&
-	    !strcmp (amodule->assembly->image->guid, amodule->image_guids [index]) &&
```
Comment (vargaz) [inline mono/mini/aot-runtime.c]: Is the mono_asmctx_get_kind (&amodule->assembly->context) == MONO_ASMCTX_INDIVIDUAL check necessary ?

Comment (lambdageek) [reply mono/mini/aot-runtime.c]: It's ok to get rid of it.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**